src/libostree/ostree-repo-pull.c

```diff
@@ -4592,7 +4592,21 @@ ostree_repo_find_remotes_async (OstreeRepo                     *self,
 
       if (local_error != NULL)
         {
-          g_warning ("Avahi finder failed; removing it: %s", local_error->message);
+          /* See ostree-repo-finder-avahi.c:ostree_repo_finder_avahi_start, we
+           * intentionally throw this so as to distinguish between the Avahi
+           * finder failing because the Avahi daemon wasn't running and
+           * the Avahi finder failing because of some actual error.
+           *
+           * We need to distinguish between g_debug and g_warning here because
+           * unit tests that use this code may set G_DEBUG=fatal-warnings which
+           * would cause client code to abort if a warning were emitted. See
+           * https://phabricator.endlessm.com/T16682#530304
```

Comment:
(I realise this is not what I normally say, but) You can omit the Endless Phabricator link from this comment, since it won’t be useful to anyone upstream, and the comment explains things fairly well by itself.